base/shell/explorer/startctxmnu.cpp

```diff
@@ -169,7 +168,15 @@ class CStartMenuBtnCtxMenu :
                     if (SUCCEEDED(hRet))
                     {
                         CreateContextMenuFromShellFolderPidl(hPopup);
-                        m_idCmdCmLast = ID_SHELL_CMD_FIRST + GetMenuItemCount(hPopup);
+
+                        m_idCmdCmLast = m_Inner->QueryContextMenu(
```

Comment:
You should not re-query this, since it is already queried on L49! (Depending on how the shell extension is implemented, you now have double items in that menu)
Also, you are using the return value directly, which is incorrect. I explicitly said to use `HRESULT_CODE ` for that.